pkgs/merkle-tree-wasm/src/lib.rs

```diff
@@ -98,7 +98,7 @@ pub fn generate_merkle_proof(
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-pub fn generate_merkle_proof(
```

Comment:
what is wrong with keeping `generate_merkle_proof`?

`generate_merkle_proof_cli` makes an assumption about what will use this function (a CLI), but you shouldn't actually make assumptions about what/who will use your public library functions...